src/libsystemd/sd-device/device-util.c

```diff
@@ -65,6 +65,33 @@ int device_open_from_devnum(mode_t mode, dev_t devnum, int flags, char **ret) {
         return TAKE_FD(fd);
 }
 
+int device_is_removable(sd_device *device) {
```

Comment:
one more tweak, can you rename this `block_device_is_removable()`, because this is only about block devices.